react-stopwatch/my-app/src/Stopwatch.tsx

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from 'react';
+import { FaPlay } from 'react-icons/fa';
+import { FaPause } from 'react-icons/fa';
+import './Stopwatch.css';
+
+export function Stopwatch() {
+  const [isActive, setIsActive] = useState(false);
+  const [isPaused, setIsPaused] = useState(true);
+  const [time, setTime] = useState(0);
+
+  useEffect(() => {
+    let intervalId: null | NodeJS.Timeout = null;
+
+    if (isActive && isPaused === false) {
+      intervalId = setInterval(() => {
+        setTime((time) => time + 1);
+      }, 1000);
+    } else {
+      clearInterval(intervalId ?? 0);
+    }
+    return () => {
+      clearInterval(intervalId ?? 0);
```

Comment:
This is the effect "clean up function". It gets called just before the effect function itself gets called, so it will clear your interval before you create a new one. It works, but since you are always clearing it and restarting it, you could have used `setTimeout` instead.

Also note that this call here is "capturing" `intervalId`, so it will be valid when the clean up function is called. Good.